tgui/packages/tgui/interfaces/PreferencesMenu/loadout/index.tsx

```diff
@@ -0,0 +1,339 @@
+                  <LoadoutTabDisplay category={activeCategory} />
+                )}
+              </Stack.Item>
+            </Stack>
+          </Section>
+        ) : (
+          <Section fill>
+            <Box>No contents for selected tab.</Box>
+          </Section>
+        )}
+      </Stack.Item>
+    </Stack>
+  );
+};
+
+const typepathToLoadoutItem = (
+  typepath: typePath,
+  all_tabs: LoadoutCategory[],
+) => {
+  // Maybe a bit inefficient, could be replaced with a hashmap?
+  for (const tab of all_tabs) {
+    for (const item of tab.contents) {
+      if (item.path === typepath) {
+        return item;
+      }
+    }
+  }
+  return null;
+};
+
+const LoadoutSelectedItem = (props: {
+  key: string;
+  path: typePath;
+  all_tabs: LoadoutCategory[];
+  modifyItemDimmer: LoadoutItem | null;
+  setModifyItemDimmer: (dimmer: LoadoutItem | null) => void;
+}) => {
+  const { all_tabs, path, key, modifyItemDimmer, setModifyItemDimmer } = props;
+  const { act } = useBackend();
+
+  const item = typepathToLoadoutItem(path, all_tabs);
+  if (!item) {
+    return null;
+  }
+
+  return (
+    <Stack key={key} align={'center'}>
+      <Stack.Item>
+        <ItemIcon item={item} scale={1} />
+      </Stack.Item>
+      <Stack.Item width="55%">{item.name}</Stack.Item>
+      {item.buttons.length ? (
+        <Stack.Item>
+          <Button
+            color="none"
+            width="32px"
+            onClick={() => {
+              setModifyItemDimmer(item);
+            }}
+          >
+            <Icon size={1.8} name="cogs" color="grey" />
+          </Button>
+        </Stack.Item>
+      ) : (
+        <Stack.Item width="32px" /> // empty space
+      )}
+      <Stack.Item>
+        <Button
+          color="none"
+          width="32px"
+          onClick={() => act('select_item', { path: path, deselect: true })}
+        >
+          <Icon size={2.4} name="times" color="red" />
+        </Button>
+      </Stack.Item>
+    </Stack>
+  );
+};
+
+const LoadoutSelectedSection = (props: {
+  all_tabs: LoadoutCategory[];
+  modifyItemDimmer: LoadoutItem | null;
+  setModifyItemDimmer: (dimmer: LoadoutItem | null) => void;
+}) => {
+  const { act, data } = useBackend<LoadoutManagerData>();
+  const { loadout_list } = data.character_preferences.misc;
+  const { all_tabs, modifyItemDimmer, setModifyItemDimmer } = props;
+
+  return (
+    <Section
+      title="&nbsp;"
+      scrollable
+      fill
+      buttons={
+        <Button.Confirm
+          icon="times"
+          color="red"
+          align="center"
+          disabled={!loadout_list || Object.keys(loadout_list).length === 0}
+          tooltip="Clears ALL selected items from all categories."
+          onClick={() => act('clear_all_items')}
+        >
+          Clear All
+        </Button.Confirm>
+      }
+    >
+      {loadout_list &&
+        Object.entries(loadout_list).map(([path, item]) => (
+          <>
+            <LoadoutSelectedItem
+              key={path}
```

Comment:
keys are required to be on the topmost child elements within map, which in this case is the empty fragment, not loadoutselecteditem. this could actually cause issues, so you'd need to move the `key` prop to the empty <>. import and use `Fragment` to do this.